rholang/src/test/scala/coop/rchain/rholang/PeekSpec.scala

```diff
@@ -0,0 +1,62 @@
+package coop.rchain.rholang
+
+import org.scalatest._
+
+import coop.rchain.metrics.Metrics
+import coop.rchain.models.Expr.ExprInstance.{GInt, GString}
+import coop.rchain.models.rholang.implicits._
+import coop.rchain.shared.Log
+import coop.rchain.rholang.interpreter.{EvaluateResult, InterpreterUtil, Runtime}
+
+import monix.eval.Task
+import monix.execution.Scheduler.Implicits.global
+
+import scala.concurrent.duration._
+
+class PeekSpec extends FlatSpec with Matchers {
+
+  import Resources._
+
+  implicit val logF: Log[Task]            = new Log.NOPLog[Task]
+  implicit val noopMetrics: Metrics[Task] = new Metrics.MetricsNOP[Task]
+
+  val tmpPrefix = "peek-spec-"
+
+  "peek" should "not remove read data" in {
+    mkRuntime[Task](tmpPrefix)
+      .use { runtime =>
+        implicit val c = runtime.cost
+        for {
+          res1 <- InterpreterUtil.evaluate[Task](runtime, """@1!("v1") | for(_ <<- @1) { Nil }""")
+          _    = res1.errors shouldBe empty
```

Comment:
Please extract a `success` method to reduce the noise of these two tests.
Ideally, we'd have it in a util class in `rholang` tests.

Actually, InterpreterUtil should be in `test` scope - it's only used there. Please move it to test scope, then we can embed the 'no errors when evaulating' asseriton in its method. To discern between expected results, let's have two methods:

`evaulate(rho: String): F[Unit]`
`evaulateResult(rho: String): F[EvaulateResult]`

Reply:
I don't think moving assertions to "library" methods is a good idea, because when they eventually fail the reported information about the line at which this happened is useless. Following the DRY principle at all cost doesn't always make sense in tests

Reply:
> when they eventually fail the reported information about the line at which this happened is useless

It depends on how you implement them. Often it's enough to display the assertion inputs in the error message, and not have a 'true was false' assertion. In this case we should have the exact rholang line that failed.

If that's not enough for you, you can thread source position information using Li Haoyi's `sourcecode`, which we have on classpath. See `shared.Debug` for example usage.

Reply:
The assertion checks whether execution was successful, so displaying rholang alone doesn't make sense here because the result of the execution doesn't depend solely on the passed in code.

I don't see now how I could use this library for that specific purpose. Let's discuss with the team whether we want to go ahead with this approach

Reply:
> displaying rholang alone doesn't make sense

feel free to include whatever makes sense, or what the inline assertions include.

> I don't see now how I could use this library

Have you seen its [readme](https://github.com/lihaoyi/sourcecode#sourcecode---)? It's pretty self-explanatory.

Reply:
Yes, it looks pretty clear, thanks. I'm still not convinced we should manually do what `scalatest` does automatically for us.
Currently we have:

```
[info] - should not remove read data *** FAILED *** (2 seconds, 664 milliseconds)                            
[info]   Vector(coop.rchain.rholang.interpreter.errors$SyntaxError: syntax error(): @ at 1:2-1:3) was not empty (PeekSpec.scala:33)
```
After your suggestion is implemented:
```
[info] - should not remove read data *** FAILED *** (2 seconds, 606 milliseconds)
[info]   Vector(coop.rchain.rholang.interpreter.errors$SyntaxError: syntax error(): @ at 1:2-1:3) was not empty (InterpreterUtil.scala:20) 
```
One simply cannot tell which evaluation caused the assertion to fail. 

After passing in implicit `sourcecode` parameters to `evaluate` the following could be displayed, but the info at the very end to which everybody is used would probably make it confusing:
```
[info] - should not remove read data *** FAILED *** (2 seconds, 391 milliseconds)
[info]   The failure was in: /home/lukasz/git/coop/rchain/rholang/src/test/scala/coop/rchain/rholang/PeekSpec.scala:32 Vector(coop.rchain.rholang.interpreter.errors$SyntaxError: syntax error(): @ at 1:2-1:3) was not empty (InterpreterUtil.scala:24)
```

Reply:
Consider this: tests are read more often than the assertion messages themselves. Notice also there isn't really a tradeoff between the two readabilities, and I'd argue we can even do much better than the message that would be provided by the inline assertion.

Re: familiarity - each and every assertion error needs careful attention anyway. I'd start by catching the original error and rethrowing new one with source position or/and executed rholang info in the new message, and the old message as the `cause` of the new AssertionError.

Reply:
Ok, you've convinced me. I'll change this in the follow-up PR